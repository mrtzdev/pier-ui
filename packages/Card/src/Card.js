import React from "react";
import classNames from "classnames";
import PropTypes from "prop-types";
import defaultTheme from "../../Theme/theme";
import { useTheme } from "../../PierUIProvider";

const propTypes = {
  variant: PropTypes.oneOf(["flat", "outlined", "shadow"]),
  className: PropTypes.string,
  as: PropTypes.string,
  children: PropTypes.any,
  maxWidth: PropTypes.string,
  borderRadius: PropTypes.string,
};

const defaultProps = {
  variant: "flat",
  as: "div",
};

const Card = (props) => {
  const {
    variant,
    className,
    maxWidth,
    borderRadius,
    as,
    children,
    ...restProps
  } = props;
  const Component = as;

  const customTheme = useTheme();
  const theme = customTheme ? customTheme : defaultTheme;

  return (
    <>
      <Component
        className={classNames("card", className, variant && `${variant}`)}
        {...restProps}
      >
        {children}
      </Component>
      <style jsx>{`
        .card {
          width: 100%;
          position: relative;
          min-height: 100px;
          overflow: hidden;
          ${maxWidth ? "max-width:" + maxWidth : "max-width:" + "none"};
          ${borderRadius
            ? "border-radius:" + borderRadius
            : "border-radius:" + theme.borderRadius.base};
        }

        .card.flat {
          background-color: ${theme.colors.greyLight};
          border: 1px solid ${theme.colors.greyLight};
        }
        .card.outlined {
          border: 1px solid ${theme.colors.grey};
        }
        .card.shadow {
          box-shadow: 0 3px 15px -4px rgb(0 0 0 / 30%);
        }
      `}</style>
    </>
  );
};

export default Card;

Card.displayName = "Card";
Card.propTypes = propTypes;
Card.defaultProps = defaultProps;
