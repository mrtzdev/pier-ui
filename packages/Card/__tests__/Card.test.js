/**
 * @jest-environment jsdom
 */

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Card from "../index";

describe("Card", () => {
  it("should render Card correctly", () => {
    render(<Card></Card>);
    screen.debug();
  });

  it("Renders with a custom className ", () => {
    const { container } = render(<Card className="custom-class"> </Card>);
    expect(container.firstChild).toHaveClass("custom-class");
  });

  it("Renders with a className equal to the variant", () => {
    const { container } = render(<Card variant="outlined" />);
    expect(container.firstChild).toHaveClass("outlined");
  });
});
