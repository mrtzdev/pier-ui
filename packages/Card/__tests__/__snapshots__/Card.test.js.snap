// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render empty card correctly 1`] = `
<Card
  as="div"
  variant="flat"
/>
`;
